### ISSUE TITLE
Allow option to set labels on pods using `kubernetes` decorator

### ISSUE THREAD

**Issue by ShantanuKumar** [Netflix/metaflow#1880]
We would like to add labels to pod generated using `kubernetes` decorator. I see there are some [internal labels ](https://github.com/Netflix/metaflow/blob/master/metaflow/plugins/kubernetes/kubernetes.py#L406)being set by Metaflow but we would like to pass some external ones too.

**Comment by savingoyal** [comment]
yes - this has been on our backlog. we had a PR that had to be rolled back due to some issues. stay tuned.

**Comment by ggutierrez545-ai** [comment]
Would this issue be one open to contributions? I've got a use case that requires setting both `labels` and `annotations` on the jobs that Metaflow creates.

After going through the source code, I believe I know what updates would need to be made to support those enhancements. Would be happy to put up a PR and work through that, if the team is interested. Not sure if this has already been handled (as the above comment somewhat suggests).

**Comment by ShantanuKumar** [comment]
Pinging again since we also need both `label` and `annotations`

**Comment by savingoyal** [comment]
#1568 is an inflight PR - if you would like to weigh in with testing that
PR that would be great.
